spec/grover/processor_spec.rb

```diff
@@ -490,6 +490,50 @@
         it { expect(pdf_text_content).to eq "#{date} Hey there http://www.example.net/foo/bar 1/1" }
       end
 
+      context 'when raise on request failure option is specified' do
+        let(:options) { basic_header_footer_options.merge('raiseOnRequestFailure' => true) }
+        let(:date) { Date.today.strftime '%-m/%-d/%Y' }
+
+        context 'when a failure occurs it raises an error' do
+          let(:url_or_html) do
+            <<-HTML
+              <html>
+                <body>
+                  <img src="http://foo.bar/baz.img" />
+                </body>
+              </html>
+            HTML
+          end
+
+          it do
+            expect do
+              convert
+            end.to raise_error Grover::JavaScript::RequestFailedError, %r{net::ERR_NAME_NOT_RESOLVED at http://foo.bar/baz.img}
```

Comment:
Hmm you seem to have missed the test case for a 404? Ie valid host, invalid path.

Reply:
You're right! Thanks for working through this with me and turning this into a real PR. 😎